hc_dpki/src/utils.rs

```diff
@@ -0,0 +1,93 @@
+use crate::SEED_SIZE;
+use hcid::*;
+use holochain_core_types::{agent::Base32, error::HcResult};
+use holochain_sodium::{secbuf::SecBuf, sign};
+
+/// Decode an HCID-encoded key into a SecBuf
+/// @param {Base32} pub_key_b32 - Public signing key to decode
+/// @param {HcidEncoding} codec - The configured HCID decoder to use
+/// @return {SecBuf} Resulting decoded key
+pub(crate) fn decode_pub_key(pub_key_b32: Base32, codec: &HcidEncoding) -> HcResult<SecBuf> {
+    // Decode Base32 public key
+    let pub_key = codec.decode(&pub_key_b32)?;
+    // convert to SecBuf
+    let mut pub_key_sec = SecBuf::with_insecure(sign::PUBLICKEYBYTES);
+    {
+        let mut pub_key_lock = pub_key_sec.write_lock();
+        for x in 0..pub_key.len() {
+            pub_key_lock[x] = pub_key[x];
+        }
+    }
+    Ok(pub_key_sec)
+}
+
+/// Encode with HCID a public key given as a SecBuf
+/// @param {SecBuf} pub_key_sec - Public signing key to encode
+/// @param {HcidEncoding} codec - The configured HCID encoder to use
+/// @return {Base32} Resulting HCID encoded key
+pub(crate) fn encode_pub_key(pub_key_sec: &mut SecBuf, codec: &HcidEncoding) -> HcResult<Base32> {
+    let locker = pub_key_sec.read_lock();
+    let pub_buf = &*locker;
+    let pub_key = array_ref![pub_buf, 0, SEED_SIZE];
+    Ok(codec.encode(pub_key)?)
```

Comment:
Curious why the `array_ref!`? a direct reference, should work, or if not, a slice
```rust
Ok(codec.encode(&locker[0..SEED_SIZE]))
```

Reply:
No reason, just copy-paste ^^'''